tree/ntuple/v7/src/RPageStorageFile.cxx

```diff
@@ -258,18 +258,34 @@ void ROOT::Experimental::Internal::RPageSourceFile::InitDescriptor(const RNTuple
    }
 
    fDescriptorBuilder.SetOnDiskHeaderSize(anchor.GetNBytesHeader());
-   auto buffer = std::make_unique<unsigned char[]>(anchor.GetLenHeader());
-   auto zipBuffer = std::make_unique<unsigned char[]>(anchor.GetNBytesHeader());
-   fReader.ReadBuffer(zipBuffer.get(), anchor.GetNBytesHeader(), anchor.GetSeekHeader());
-   fDecompressor->Unzip(zipBuffer.get(), anchor.GetNBytesHeader(), anchor.GetLenHeader(), buffer.get());
-   RNTupleSerializer::DeserializeHeader(buffer.get(), anchor.GetLenHeader(), fDescriptorBuilder);
-
    fDescriptorBuilder.AddToOnDiskFooterSize(anchor.GetNBytesFooter());
-   buffer = std::make_unique<unsigned char[]>(anchor.GetLenFooter());
-   zipBuffer = std::make_unique<unsigned char[]>(anchor.GetNBytesFooter());
-   fReader.ReadBuffer(zipBuffer.get(), anchor.GetNBytesFooter(), anchor.GetSeekFooter());
-   fDecompressor->Unzip(zipBuffer.get(), anchor.GetNBytesFooter(), anchor.GetLenFooter(), buffer.get());
-   RNTupleSerializer::DeserializeFooter(buffer.get(), anchor.GetLenFooter(), fDescriptorBuilder);
+   auto headerBuf = std::make_unique<unsigned char[]>(anchor.GetNBytesHeader());
```

Comment:
We could further optimize by doing a single allocation of combined size for header + footer + unzip buffers, since we're throwing away these buffers at the end of the scope.